### PR TITLE
To resolve the conflict

### PR DESCRIPTION
**Pull request by shanlihou**
this commit has merge the bookmark-tree to main

### CODE REVIEW

File: lua/bookmarks/domain/bookmark_list.lua [hunk 1]
```diff
@@ -1,11 +1,12 @@
-local bookmark_scope = require("bookmarks.domain.bookmark")
+local M = require("bookmarks.domain.bookmark")
```
Comment (LintaoAmons) [inline lua/bookmarks/domain/bookmark_list.lua:1]: redefined local `M`.  line 1 and line 11

**Review by LintaoAmons — COMMENTED** [review]
LGTM, there's only some minor changes you may need take a look

File: lua/bookmarks/domain/bookmark_list.lua (hunk 1 above)
Comment (shanlihou) [reply lua/bookmarks/domain/bookmark_list.lua:1]: oh, after merge, I replace all the bookmarks list name to M

File: lua/bookmarks/domain/bookmark_list.lua [hunk 2]
```diff
@@ -33,9 +33,16 @@ end
 function M.find_bookmark_by_location(self, location)
   -- TODO: self location path
   for _, b in ipairs(self.bookmarks) do
+    local b_type = M.get_value_type(b)
```
Comment (LintaoAmons) [inline lua/bookmarks/domain/bookmark_list.lua:36]: Seems like this commit will fail the sign and line highlight.
And also the original pick method is not working as well

Comment (shanlihou) [reply lua/bookmarks/domain/bookmark_list.lua:36]: is there any exception

**Review by LintaoAmons — COMMENTED** [review]
What's the difference between `tree.api` and `tree.operate`?
Are they serve for different purposes?

---
edit:
Ok, my understanding is 
`operate` is treeview buffer local,
`api` exposed to the outside as one plugin api

**Pull request merged**